parts/k8s/cloud-init/artifacts/cse_config.sh

```diff
@@ -165,14 +165,19 @@ configureK8s() {
     chmod 0644 "${APISERVER_PUBLIC_KEY_PATH}"
     chown root:root "${APISERVER_PUBLIC_KEY_PATH}"
 
-    AZURE_JSON_PATH="/etc/kubernetes/azure.json"
-    touch "${AZURE_JSON_PATH}"
-    chmod 0600 "${AZURE_JSON_PATH}"
-    chown root:root "${AZURE_JSON_PATH}"
-
-    set +x
     echo "${KUBELET_PRIVATE_KEY}" | base64 --decode > "${KUBELET_PRIVATE_KEY_PATH}"
     echo "${APISERVER_PUBLIC_KEY}" | base64 --decode > "${APISERVER_PUBLIC_KEY_PATH}"
+    configureKubeletServerCert
+    AZURE_JSON_PATH="/etc/kubernetes/azure.json"
+    if [[ -n "${MASTER_NODE}" ]]; then
+        if [[ "${ENABLE_AGGREGATED_APIS}" = True ]]; then
+            generateAggregatedAPICerts
+        fi
+    else
+        {{- /* If we are a node vm then we only proceed w/ local azure.json configuration if cloud-init has pre-paved that file */}}
```

Comment:
To be clear, we are *not* pre-paving `azure.json` for master vms (note that there is no diff to `masternodecustomdata.yml`. The existing code (specifically `cat << EOF > "${AZURE_JSON_PATH}"`) works as-is whether or not the file is there (the `> "${AZURE_JSON_PATH}"` part). So we are just using the pre-existence of `azure.json`, in a non-master flow only, to determine whether or not we will include the `azure.json` on this node. If the "check for file" test fails, we return immediately, because that tells us we aren't including `azure.json` on this node.

This works because:

1) we are paving `azure.json` on nodes, if needed, prior to paving *this* file itself (`cse_config.sh` in code, and `/opt/azure/containers/provision_configs.sh` on the local vm fs)
2) we are waiting for *this* file in `cse_main.sh` before executing code in *this* file, and because the cloud-init `write_files` property is a yaml sequence, those will be processed in order. Thus, by the time we are in *this* execution flow, *if* `azure.json` should be here, it will be here.